src/datachain/lib/pytorch.py

```diff
@@ -85,51 +92,58 @@ def _get_catalog(self) -> "Catalog":
         wh = wh_cls(*wh_args, **wh_kwargs)
         return Catalog(ms, wh, **self._catalog_params)
 
-    def __iter__(self) -> Iterator[Any]:
-        if self.catalog is None:
-            self.catalog = self._get_catalog()
-        session = Session.get(catalog=self.catalog)
-        total_rank, total_workers = self.get_rank_and_workers()
+    def _rows_iter(self, total_rank: int, total_workers: int):
+        catalog = self._get_catalog()
+        session = Session("PyTorch", catalog=catalog)
```

Comment:
We need to create a new session here, since the `AsyncMapper` runs this on a separate thread.